src/components/ActionButton/ActionButton.ts

```diff
@@ -62,7 +62,16 @@ export class ActionButton extends Component {
          * <button class='CoveoActionButton' data-icon='&lt;svg width=&quot;1em&quot; height=&quot;1em&quot;&gt;...&lt;/svg&gt;'></button>
          * ```
          */
-        icon: ComponentOptions.buildStringOption()
+        icon: ComponentOptions.buildStringOption(),
+
+        /**
+         * Specifies the handler called when the button is clicked.
+         *
+         * Default is `null`.
+         *
+         * This option must be set in JavaScript when initializing the component.
+         */
+        click: ComponentOptions.buildCustomOption(s => null)
```

Comment:
It's been a few since I used buildCustomOption, but ain't there anything to precise that there's no default value and that it **must** be set?

Reply:
You are perfectly right, I was not aware of `IComponentOptions.required`. I was able to remove the condition and warning to use that instead.